techdocs/script/detectors.py

```diff
@@ -201,100 +202,143 @@ def __init__(self, bundler=None, validator=None):
-            if isinstance(file, dict) and file.get("openapi") and file.get("paths"):
+
+        for json_op in json_ops:
+            try:
+                data = json.loads(fs.read_string(json_op.source_abs))
+            except json.JSONDecodeError:
+                continue  # Not valid JSON, skip
+
+            if isinstance(data, dict) and data.get("openapi") and data.get("paths"):
+                visited = set()
+                ref_files = self._collect_references(fs, json_op.source_abs, visited)
                 openapi_spec_files.append(
-                    OpenAPIFile(
-                        json_file.source_abs,
-                        json_file.destination_abs,
-                        self._detect_referenced_files_json(fs, json_file),
-                    )
+                    OpenAPIFile(json_op.source_abs, json_op.destination_abs, ref_files)
                 )
         return openapi_spec_files
 
-    @staticmethod
-    def _detect_referenced_files_json(fs, openapi_file):
-        ref_files = []
-        data = json.loads(fs.read_string(openapi_file.source_abs))
+    def _collect_references(self, fs, file_abs, visited):
+        if file_abs in visited:
+            return []
+        visited.add(file_abs)
 
-        def look_for_files_in_refs(obj):
+        if file_abs.lower().endswith((".yaml", ".yml")):
+            return self._collect_yaml_references(fs, file_abs, visited)
+        elif file_abs.lower().endswith(".json"):
+            return self._collect_json_references(fs, file_abs, visited)
+        else:
+            return []
+
+    def _collect_yaml_references(self, fs, file_abs, visited):
+        content = fs.read_string(file_abs)
+        lines = content.split("\n")
+        base_dir = path.dirname(file_abs)
+        abs_refs = []
+
+        for line in lines:
+            if "$ref" in line:
+                match = re.search(r"\$ref:\s*[\"']?([.a-zA-Z0-9_\-/.]+)", line)
+                if match:
+                    ref_path = match.group(1)
+                    ref_abs = path.abspath(path.join(base_dir, ref_path))
+                    abs_refs.append(ref_abs)
+
+        all_nested = []
+        for ref in abs_refs:
+            all_nested.extend(self._collect_references(fs, ref, visited))
+
+        return list(set(abs_refs + all_nested))
+
+    def _collect_json_references(self, fs, file_abs, visited):
+        try:
+            data = json.loads(fs.read_string(file_abs))
+        except json.JSONDecodeError:
+            return []
+
+        base_dir = path.dirname(file_abs)
+        collected = []
+
+        def look_for_refs(obj):
             if isinstance(obj, dict):
                 for key, value in obj.items():
-                    if key == "$ref":
-                        match = re.search(r"([.a-z_\-/]+)", value)
+                    if key == "$ref" and isinstance(value, str):
+                        match = re.search(r"^([.a-zA-Z0-9_\-/]+)", value)
```

Comment:
Why was this regex changed?

Reply:
Consider following values:
```
- ../schema/common_types.json#/definitions/uuid
- #/definitions/uuid
```
Without explicitly saying that we're interested in occurences at the start of the string it could take `/definitions/uuid` as match. Obviously there's no such file. Uppercase letters are only for more wide filename handling.

It would